packages/ra-ui-materialui/src/input/NullableBooleanInput.tsx

```diff
@@ -70,20 +77,6 @@ const NullableBooleanInput: FunctionComponent<
         validate,
     });
 
-    const enhancedOptions = displayNull
-        ? {
-              ...options,
-              SelectProps: {
-                  displayEmpty: true,
```

Comment:
The property is named `allowEmpty` so this has no effect. This component is already adding a "null" option.